README.md

```diff
@@ -47,81 +73,376 @@ The following sections specify the required and optional content for `eaw.modinf
+
+**Data Semantics**: Ordered List of Objects 
+
+**Description:**
+
+The `dependencies` container holds an ordered list of [`"modreference type"`](#the-modreference-type)s this mod relies on.
 
-The `dependencies` container holds an ordered list of other mods instances this mod relies on.
-The first item of the list is the most base mod, every `n+1` mod is an sub mod of `n`. The mod of the this modinfo.json file must not be listed here!
 The list is either absent from the `modinfo.json` or contains at least one item.
 
-#### The `"dependencies.modtype"` Tag
+The first mod in the list resembles the closest dependeny. Every `n+1` item is another dependeny of the target mod.
+If a dependency has dependencies itself these **must not** be added to the list.
+
+**The mod of the this modinfo.json file must not be listed here because that would result in a cycle!**
+
+Here is an example what it means when the list contains three mod references:
+Imagine this mod is called `A`. 
+The dependencies list contains the mods `[C, D, B]`. 
+
+In Java Syntax this is equivalent to: `A extends C, D, B`. The mod dependency graph looks as follows:
+
+```
+  C
+ /
+A - D
+ \
+  B
+```
+
+
+### The `"steamdata"` Property
+
+**Level:** *OPTIONAL*
+
+**Data Type**: [`steamdata`](#tthe-steamdata-type)  
+
+**Data Semantics**: Steam Workshops JSON
+
+**Description:**
+
+The [`"steamdata type"`](#the-steamdata-type) container holds additional info that is required for the Steam Version of the game.
+The container is either absent from the `modinfo.json` or it is fully required.
+
+### The `"custom"` Property
+
+**Level:** *OPTIONAL*
+
+**Data Type**: `Object[]`
+
+**Data Semantics**: A list of anything
+
+**Description:**
+
+
+The `custom` property allows arbitrary extensions to the format. Programs implementing the core format should always be able to serialize the whole object, but the custom data wrapped within the extension object has to be accounted for only where applicable and/or needed.
+
+---
+
+## The `"modreference"` Type
+
+#### The `"modreference.modtype"` Property
+
+**Level:** **REQUIRED**
+
+**Data Type**: `integer`
+
+**Data Semantics**: Enum
+
+**Description:**
 
 The modtype enumeration:
 
 | Value | Meaning |
 |:--:|:--|
-|0|any normal mod inside the Mods/ directory|
-|1|a Steam Workshops mod|
+|`0`|any normal mod inside the Mods/ directory|
+|`1`|a Steam Workshops mod|
+|`2`|a "virtual" mod. (Currently unsupported)|
+
+*Reasonable: The current mod does NOT contain a `modtype` property because the mod should not know it's owen type. Otherwise sharing this file across steam and disk mods would not be possible. A `modreference` requests this data, meaning tool support to determine the actual `modtype` is necessary. This design coice was made because mod linking should always be considered for Steam Workshop mods. For convenient test for mod creators we keep the possibility to reference to local mods.*
```

Comment:
```suggestion
*Rationale: The current mod does NOT contain a `modtype` property because the mod should not have to know it's own type. Otherwise sharing this file across steam and disk mods would not be possible. A `modreference` requests this data, meaning tool support to determine the actual `modtype` is necessary. This design coice was made because mod linking should always be considered for Steam Workshop mods. The possibility to reference local mods is a convenience functionality intended to be used by mod developers for test setups and development.*
```